### ISSUE TITLE
Scrolling through "Activities" crashes app.

### ISSUE THREAD

**Issue by OlivierS1** [nextcloud/android#1074]
### Actual behaviour
-Tell us what happens
Selecting "Activities", the page will be refreshed with the list of recent activities.
When I try to scroll toward the bottom of the list, the application will display the following error message before it quits:

"Unfortunately, Nextcloud dev has stopped"

### Expected behaviour
-Tell us what should happen
No crash.
 
### Steps to reproduce
1. Select "Activities".
2. When list is available, scroll the list.
3. 

### Environment data
Android version:
6.0
Device model: 
LG 3 LG-852
Stock or customized system:
Stock

Nextcloud app version:
Nextcloud Dev 20170608

Nextcloud server version:
NC 12.0

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by AndyScherzinger** [comment]
Hi @OlivierS1 do you see any StackTraces in the log file of the app?

**Comment by OlivierS1** [comment]
@AndyScherzinger , if you want to search for the word "StackTraces" in the logs, then, no. I have not found any instances of it.
As far what I see in the log, nothing obvious indicating this is Nextcloud server related.
I just suspect some memory leaks or pointer issue in the scrolling of the app page.
Also, when updating the app (now running dev 20170609), the "Activities" page remain blank. The log seems to agree with the fact, meaning the log doesn't show activities.
However, running with Dev 20170609 and restarting the app, I got the following output in the new log:

2017/06/09 08:26:22
GetRemoteActivitiesOperation : Exception while getting remote activities Exception : [com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:224), com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41), com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82), com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61), com.google.gson.Gson.fromJson(Gson.java:887), com.google.gson.Gson.fromJson(Gson.java:952), com.owncloud.android.lib.resources.activities.GetRemoteActivitiesOperation.parseResult(GetRemoteActivitiesOperation.java:129), com.owncloud.android.lib.resources.activities.GetRemoteActivitiesOperation.run(GetRemoteActivitiesOperation.java:96), com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136), com.owncloud.android.ui.activity.ActivitiesListActivity$3.run(ActivitiesListActivity.java:205), java.lang.Thread.run(Thread.java:818)]

2017/06/09 08:26:22
ActivitiesListActivity : Unexpected exception

**Comment by AndyScherzinger** [comment]
@mario any idea, seems to be related to the Json processing 

@OlivierS1 the "exception" lines are what we call a stack trace in Java :) So what you posted is what I was hoping for :+1:

**Comment by tobiasKaminsky** [comment]
@OlivierS1 is this still happening for you on the latest dev version?
I tried it with 20170627 and my server with lots of activities and there was no problem.

**Comment by OlivierS1** [comment]
@tobiasKaminsky I am now able to scroll through "Activities" page with Nextcloud Dev 20170630, however, after doing it a few times as well as interacting with the app, it started to fail again. This was yesterday.
Today, I am able to scroll through it again.
While writing this post, I have played with the feature and it crashed once more. 
So the behavior is definitely inconsistent.
I have noticed some of the thumbnails in the "Activities" page start to appear.
I clicked on one of the link within the page and it calls the picture which is not related in anyway.
For some reason it shows the very same picture which belongs to the account used with the "Nextcloud app 1.4.3"
I should issue a bug ticket for it.
